.pre-commit-config.yaml

```diff
@@ -19,13 +19,25 @@ repos:
       - id: check-merge-conflict
       - id: detect-private-key
 
-  - repo: https://github.com/pre-commit/mirrors-mypy
-    rev: v0.950
-    hooks:
-      - id: mypy
-        exclude: ^(docs/|setup\.py|tests/)
+  # - repo: local
+  #   hooks:
+  #     - id: mypy
+  #       name: mypy
+  #       entry: "mypy"
+  #       language: system
+  #       types: [python]
+  #       require_serial: true
+  #       verbose: false
+  #       exclude: ^(docs/|setup\.py)
+  #     - id: bandit
+  #       name: bandit
+  #       entry: "bandit"
+  #       language: system
+  #       types: [python]
+  #       require_serial: true
+  #       verbose: false
 
   - repo: https://github.com/pycqa/flake8
-    rev: 4.0.1
+    rev: 7.1.1
     hooks:
     -   id: flake8
```

Comment:
you can add an extra line to the pre-commit configuration for flake8 to get it to read your settings from `pyproject.toml`. It looks like the ci-cd is failing due to a few line-length errors that would otherwise be ignored if it read the config.

```yaml
-   id: flake8
    additional_dependencies: [flake8-pyproject]
```
reference: https://flake8.pycqa.org/en/latest/user/using-hooks.html